code/core/system/proxy-velodyne16/src/velodyne16Decoder.cpp

```diff
@@ -354,14 +371,12 @@ void Velodyne16Decoder::nextString(const string &payload) {
                             if (m_distanceEncoding == 0) {
                                 distance = distance / 5;
                             }
-                            distance = distance >> m_numberOfBitsForIntensity; ////Reserve the upper n bits for intensity
-                            
-                            //Map the original 256 intensity levels to 2^n levels using n bits and 
-	                        //place these n bits as the highest n bits of the byte representing both intensity and distance
-	                        uint16_t intensityLevel = thirdByte;
-	                        intensityLevel = (intensityLevel >> (8 - m_numberOfBitsForIntensity)) << (16 - m_numberOfBitsForIntensity);
-	                        m_16SensorsWithIntensity[sensorID] = intensityLevel | distance; ////n bits for intensity + (16-n) bits for distance
-                            
+                            uint16_t mask = 0xFFFF - static_cast<uint16_t>(pow(2.0, static_cast<float>(m_numberOfBitsForIntensity)) - 1);
```

Comment:
pow(...) is a very expensive call; as the mask won't change, this value can be computed during the setUp(...) call.